modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/ContentDashboardItemFactoryTracker.java

```diff
@@ -67,20 +90,174 @@ protected void activate(BundleContext bundleContext) {
 				bundleContext, ContentDashboardItemFactory.class, null,
 				ServiceReferenceMapperFactory.create(
 					bundleContext,
-					(contentDashboardItem, emitter) -> emitter.emit(
+					(contentDashboardItemFactory, emitter) -> emitter.emit(
 						GenericUtil.getGenericClassName(
-							contentDashboardItem))));
+							contentDashboardItemFactory))),
+				new ContentDashboardItemFactoryTrackerCustomizer(
+					bundleContext));
 	}
 
 	@Deactivate
 	protected void deactivate() {
 		_serviceTrackerMap.close();
 	}
 
+	private void _addAssetVocabulary(
+			Company company, long classNameId, String name, int visibilityType)
+		throws Exception {
+
+		User defaultUser = company.getDefaultUser();
+
+		Map<Locale, String> titleMap = new HashMap<>();
+
+		for (Locale locale :
+				_language.getCompanyAvailableLocales(company.getCompanyId())) {
+
+			titleMap.put(locale, LanguageUtil.get(locale, name));
+		}
+
+		AssetVocabularySettingsHelper assetVocabularySettingsHelper =
+			new AssetVocabularySettingsHelper();
+
+		assetVocabularySettingsHelper.setClassNameIdsAndClassTypePKs(
+			new long[] {classNameId},
+			new long[] {AssetCategoryConstants.ALL_CLASS_TYPE_PK},
+			new boolean[] {false});
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAddGuestPermissions(true);
+
+		_assetVocabularyLocalService.addVocabulary(
+			null, defaultUser.getUserId(), company.getGroupId(), name,
+			StringPool.BLANK, titleMap, Collections.emptyMap(),
+			assetVocabularySettingsHelper.toString(), visibilityType,
+			serviceContext);
+	}
+
+	private void _addClassNameIdToVocabularyIfNotExist(
+		long classNameId, String name) {
+
+		try {
+			_companyLocalService.forEachCompany(
+				company -> {
+					AssetVocabulary assetVocabulary =
+						_assetVocabularyLocalService.fetchGroupVocabulary(
+							company.getGroupId(),
```

Comment:
Just a doubt about this `groupId` :
We've concluded that because we can have nested companies, forming a "group", this way we are fetching all vocabularies from a company, those that are directly assigned and the ones that the company is inheriting from a possible parent company. Are we right about this?

Reply:
This is the default goupId of the company. What do you mean as parent company? Here you can check all the info of how Virtual Instances work in Liferay: https://help.liferay.com/hc/en-us/articles/360017877332-Virtual-Instances

Reply:
Does this `groupId` refer to User group?